File: JavaLab/BinaryToOctal.java
import java.util.Scanner;
class BinaryToOctal
{
	static public void main(String args[])
	{
		Scanner sc=new Scanner(System.in);
		System.out.println("Enter Binary Digits:");
		int binary=sc.nextInt();
		int rem,dec=0,i=0;
		while(binary>0)
		{
			rem=binary%10;
			dec+=Math.pow(2,i)*rem;
			binary=binary/10;
			i++;
		}
		//System.out.println(dec);
		int digits=dec;
		int r,y=1,sum=0;
		while(digits>8)
		{
			r=digits%8;
			sum+=r*y;
			y=y*2;
			digits/=10;
		}
		System.out.println(sum);
	}

}

File: CLab/BinaryToOctal.c
#include"math.h"
#include"stdio.h"
void main()
{
	int num, dec=0, rem,i=0;
	printf("Enter Binary Digits:");
	scanf("%d",&num);
	while(num!=0)
	{
		rem=num%10;
		dec+=pow(2,i)*rem;
		num=num/10;
		i++;
	}
	printf("%d",dec);
}
